### PR TITLE
Remove ClientAllocation Object

### PR DESCRIPTION
**Pull request by Kishan-Dhakan**
- Updates error messages for relevant tests

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | remove-client-allocation  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | remove-client-allocation  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

File: tests/cli_tests/zboxcli_update_allocation_test.go [hunk 1]
```diff
@@ -397,7 +397,7 @@ func TestUpdateAllocation(t *testing.T) {
 		require.NotNil(t, err, "expected error updating "+
 			"allocation", strings.Join(output, "\n"))
 		require.Len(t, output, 1)
-		reg := regexp.MustCompile(`Error updating allocation:allocation_updating_failed: can't find allocation in client's allocations list: [a-z0-9]{64} \(1\)`)
+		reg := regexp.MustCompile(`Error updating allocation:allocation_updating_failed only owner can update the allocation`)
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_update_allocation_test.go:400]: this message seems wrong - we're attempting to update an expired allocation which is owned by the current wallet

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | remove-client-allocation  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | remove-client-allocation  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

File: tests/cli_tests/zboxcli_update_allocation_test.go (hunk 1 above)
Comment (Kishan-Dhakan) [reply tests/cli_tests/zboxcli_update_allocation_test.go]: No this case is `"Update Other's Allocation Should Fail"`, so this error seems valid. The one you mentioned is above on line 283 with error message `Error updating allocation:allocation_updating_failed: can't update expired allocation` which seems correct too

File: tests/cli_tests/zboxcli_update_allocation_test.go [hunk 2]
```diff
@@ -277,7 +277,7 @@ func TestUpdateAllocation(t *testing.T) {
 
 		require.NotNil(t, err, "expected error updating allocation", strings.Join(output, "\n"))
 		require.True(t, len(output) > 3, "expected output length be at least 4", strings.Join(output, "\n"))
-		require.Equal(t, "Error updating allocation:allocation_updating_failed: can't find allocation in client's allocations list: 123abc (0)", output[3])
+		require.Equal(t, "Error updating allocation:allocation_updating_failed: can't get existing allocation: value not present", output[3])
```
[the remark this answers is not in the document]
Comment (Kishan-Dhakan) [reply tests/cli_tests/zboxcli_update_allocation_test.go:280]: Currently there's an object in 0chain code `ClientAllocations` which stores all allocations a client owns. This served two purposes:
1: Fetch a list of all allocations a client owns from MPT.
2: Check whether any request regarding allocation comes from it's owner (by checking if allocation exists in that client's clientallocation).

1 is now done through eventsDB with a query, and 2 can be done by checking if t.ClientID == alloc.Owner and request.OwnerID == alloc.Owner. This object clientallocation which can be of unbounded size is not necessary anymore so it's removed as part of my PR https://github.com/0chain/0chain/pull/1490.

So earlier the error used to say this allocation does not exist in this clien't list, but now there is no list, so it says can't find allocation

File: tests/cli_tests/zboxcli_update_allocation_test.go (hunk 1 above)
Comment (stewartie4) [reply tests/cli_tests/zboxcli_update_allocation_test.go]: silly me, mobile app made it difficult to read!

File: tests/cli_tests/zboxcli_update_allocation_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (stewartie4) [reply tests/cli_tests/zboxcli_update_allocation_test.go:280]: thanks for the context

**Review by stewartie4 — APPROVED** [review]

**Pull request merged**